daemon/CryptoImpl/crypto_p.h

```diff
@@ -391,6 +391,32 @@ public Q_SLOTS:
             Sailfish::Crypto::Result &result,
             QByteArray &decrypted);
 
+    void authenticatedEncrypt(
```

Comment:
After more thought, I'm not convinced that this should be a separate method.  It's still encryption and decryption, using the same algorithms as anything else, only the mode is different (and some different parameters required).  We can document those parameter differences in the request object, so not sure having separate methods makes sense.

Reply:
Originally I tried just extending the encrypt() method and the code became convoluted in checking the different paths and conditions, so it seemed better to just separate the functionality. The explicit separation in the API seems more clear to the user as well, instead of using parameter checking to determine the correct operation. 

For the D-Bus method, the authenticatedEncrypt() returns an additional parameter (the tag) so that one can't be merged with encrypt(), unless encrypt() is also changed to return additional (empty) values.